### PR TITLE
RDSv2: Add Virtual hosts and route configs to rds code

### PR DESCRIPTION
**Pull request by rshriram**
This completes some missing functionality in existing RDS code, by generating virtual hosts,
and route configurations per service port. A couple of utility functions have been added to
allow building inbound http connection managers easily.

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

File: pilot/pkg/networking/v1alpha3/httproute.go [hunk 1]
```diff
@@ -65,6 +64,9 @@ type GuardedHost struct {
 
 	// Routes in the virtual host
 	Routes []GuardedRoute
+
+	// Hack: RouteConfiguration for this port
+	RouteConfiguration *v2.RouteConfiguration
```
Comment (kyessenov) [inline pilot/pkg/networking/v1alpha3/httproute.go]: This is wrong. Why are you stuffing route configuration here?

File: pilot/pkg/networking/v1alpha3/httproute.go [hunk 2]
```diff
@@ -231,10 +268,10 @@ type GuardedRoute struct {
 	Gateways []string
 }
 
-// TranslateRoutes creates virtual host routes from the v1alpha3 config.
+// TranslateHTTPRoutes creates virtual host routes from the v1alpha3 config.
 // The rule should be adapted to destination names (outbound clusters).
 // Each rule is guarded by source labels.
-func TranslateRoutes(in model.Config, name ClusterNaming) []GuardedRoute {
+func TranslateHTTPRoutes(in model.Config, name ClusterNaming) []GuardedRoute {
```
Comment (kyessenov) [inline pilot/pkg/networking/v1alpha3/httproute.go]: Why are you renaming this? There is no other routes but HTTP in envoy.

File: pilot/pkg/networking/v1alpha3/httproute.go [hunk 3]
```diff
@@ -452,3 +489,71 @@ func TranslateTime(in *types.Duration) *time.Duration {
 	}
 	return &out
 }
+
+// buildDefaultHTTPRoute builds a default route.
+func buildDefaultHTTPRoute(clusterName string) *route.Route {
```
Comment (kyessenov) [inline pilot/pkg/networking/v1alpha3/httproute.go]: Remove this. This is old crap code from v1.

File: pilot/pkg/networking/v1alpha3/httproute.go [hunk 4]
```diff
@@ -452,3 +489,71 @@ func TranslateTime(in *types.Duration) *time.Duration {
...
+		instance.Service.Hostname, instance.Endpoint.ServicePort)
+	defaultRoute := buildDefaultHTTPRoute(clusterName)
+
+	inboundVHost := route.VirtualHost{
+		Name:    fmt.Sprintf("%s|http|%d", model.TrafficDirectionInbound, instance.Endpoint.ServicePort.Port),
+		Domains: []string{"*"},
+		Routes:  []route.Route{*defaultRoute},
+	}
+
+	// TODO: mixer disabled for now as its configuration is still in old format
```
Comment (kyessenov) [inline pilot/pkg/networking/v1alpha3/httproute.go]: No, this is not OK just to comment out the major piece of istio functionality.

File: pilot/pkg/networking/v1alpha3/httproute.go [hunk 5]
```diff
@@ -452,3 +489,71 @@ func TranslateTime(in *types.Duration) *time.Duration {
...
+			Value: false,
+		},
+	}
+}
+
+func last(arr []string) string {
+	return arr[len(arr)-1]
+}
+
+func reverse(s string) string {
```
Comment (kyessenov) [inline pilot/pkg/networking/v1alpha3/httproute.go]: wtf is this?

**Comment by ijsnellf** [comment]
I think the new stuff for building the route config shouldn't need to modify the existing code.

**Comment by rshriram** [comment]
I am finishing up code that is deceptively in TBD form. No virtual hosts, no inbound http connection manager, etc. If you would like to finish this yourself without complaining, you are more than welcome to.

**Comment by kyessenov** [comment]
I'm going to complain, since this change has no tests, no comments, and no
design doc. This is disruptive to the plan we agreed upon, and I don't see
the point of shuffling all the inactive code.

**Pull request closed without merging**